internal/provider/validate.go

```diff
@@ -170,55 +166,51 @@ func (v *validator) validate(ctx context.Context) error {
 	err = check.MixedDirectoriesCheck(files)
 	result = errors.Join(result, err)
 
-	if dirExists(filepath.Join(v.providerDir, "docs")) {
+	if dirExists(v.providerFS, "docs") {
 		v.logger.infof("detected static docs directory, running checks")
-		err = v.validateStaticDocs(filepath.Join(v.providerDir, "docs"))
+		err = v.validateStaticDocs()
 		result = errors.Join(result, err)
 
 	}
-	if dirExists(filepath.Join(v.providerDir, filepath.Join("website", "docs"))) {
+	if dirExists(v.providerFS, "website/docs") {
```

Comment:
Yes, as far as the internal usages go, we should be using forward slashes instead of `filepath.Join`. For the error messages though, whenever we print out the path, we convert the slashes back to the OS separator so that it isn't confusing to Windows developers. So the test assertions should still use `filepath.Join` (this is also why the testscript tests are skipped on Windows).